### PR TITLE
Ap photogram ajax

### PR DESCRIPTION
**Pull request by APatel-AI**

### CODE REVIEW

File: app/views/comments/edit.html.erb
Comment (MayaS1111) [inline app/views/comments/edit.html.erb:1]: I think you might need this....

File: app/views/follow_requests/_follow_unfollow.html.erb [hunk 1]
```diff
@@ -1,17 +1,19 @@
+<div class = "<%= dom_id(sender)%>_<%=dom_id(recipient)%>_follow_unfollow">
```
Comment (MayaS1111) [inline app/views/follow_requests/_follow_unfollow.html.erb:1]: Good job for having both the sender and recipient and using "class"

File: app/views/follow_requests/create.js.erb
Comment (MayaS1111) [inline app/views/follow_requests/create.js.erb:1]: Cool idea adding using vars

File: app/views/likes/_form.html.erb [hunk 2]
```diff
@@ -1,11 +1,10 @@
-<%= form_with(model: like, class: "d-inline-block") do |form| %>
-  <%= form.hidden_field :fan_id %>
+<%= form_with(model: like, class: "d-inline-block",
+    data: {remote: true}) do |form| %>  <%= form.hidden_field :fan_id %>
```
Comment (MayaS1111) [inline app/views/likes/_form.html.erb]: Dry up code. Use "local"

File: app/views/likes/create.js.erb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+var currentLikes = $("#<%= dom_id(@like.photo)%>_likes")
+
+var updatedLikes = $("<%= j(render "photos/likes", photo: @like.photo)%>");
+
+currentLikes.replaceWith(updatedLikes);
+
+var icon = $("#<%= dom_id(@like.photo)%>_likes_icon");
```
Comment (MayaS1111) [inline app/views/likes/create.js.erb]: Not really needed, but cool

File: app/views/likes/destroy.js.erb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+var currentLikes = $("#<%= dom_id(@like.photo)%>_likes")
+
+var updatedLikes = $("<%= j(render "photos/likes", photo: @like.photo)%>");
+
+currentLikes.replaceWith(updatedLikes);
+
+var icon = $("#<%= dom_id(@like.photo)%>_likes_icon");
```
Comment (MayaS1111) [inline app/views/likes/destroy.js.erb]: Same here

**Review by MayaS1111 — APPROVED** [review]
Good job, but maybe make the small changes

File: app/views/comments/create.js.erb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+
```
Comment (ahuynh3a) [inline app/views/comments/create.js.erb]: please remove any unnecessary white space

File: app/views/comments/edit.js.erb [hunk 6]
```diff
@@ -0,0 +1,3 @@
+var edited_comment = ("<%= j(render 'form', comment: @comment) %>");
```
Comment (ahuynh3a) [inline app/views/comments/edit.js.erb]: Is this just 'form' or is it 'comment/form'

File: app/views/comments/_form.html.erb
Comment (ahuynh3a) [inline app/views/comments/_form.html.erb:1]: You are missing a unique id for the form here, which you will need to call in your update.js.erb

File: app/views/comments/update.js.erb [hunk 7]
```diff
@@ -0,0 +1 @@
+$("#<%= dom_id(@comment) %>").replaceWith("<%= j(render @comment) %>");
```
Comment (ahuynh3a) [inline app/views/comments/update.js.erb]: hmm this isn't quite right. You are trying to replace the comment in the form so you must pull the unique id from the comments/_form (which you didn't create), let me know if you have questions about this!

File: app/views/likes/_form.html.erb [hunk 8]
```diff
@@ -1,11 +1,10 @@
-<%= form_with(model: like, class: "d-inline-block") do |form| %>
-  <%= form.hidden_field :fan_id %>
+<%= form_with(model: like, class: "d-inline-block",
```
Comment (ahuynh3a) [inline app/views/likes/_form.html.erb]: this should just be remote: true

**Review by ahuynh3a — CHANGES_REQUESTED** [review]
Hi Arpan! You are off to a great start! Please make the changes accordingly and I can review/approve your pr! Please let me know if you have any questions!

File: app/views/follow_requests/create.js.erb
Comment (dylansarikas) [inline app/views/follow_requests/create.js.erb:1]: Nice job creating for variables for this!  It makes the code much more readable!

**Review by dylansarikas — APPROVED** [review]
Looks good!

**Review by ahuynh3a — APPROVED** [review]
Thank you for making the changes!

**Review by FadiBaker92 — COMMENTED** [review]
Great job using variables, it makes the code easy to read. Looks good!

**Pull request merged**